ethereum/p2p/src/test/java/org/hyperledger/besu/ethereum/p2p/discovery/internal/PeerDiscoveryControllerTest.java

```diff
@@ -1371,6 +1371,45 @@ public void shouldNotRespondToENRRequestForNonBondedPeer() {
         .send(eq(peers.get(0)), matchPacketOfType(PacketType.ENR_REQUEST));
   }
 
+  @Test
+  public void shouldNotRespondAndCacheENRRequestForBondingPeer() {
```

Comment:
This test name is a bit confusing to me, what about:
```suggestion
  public void shouldRespondToRecentENRRequestAfterBonding() {
```